django/middleware/http.py

```diff
@@ -19,6 +19,9 @@ def process_response(self, request, response):
         if not response.streaming and not response.has_header('Content-Length'):
             response['Content-Length'] = str(len(response.content))
 
+        if request.method not in ('GET', 'HEAD'):
+            return response
+
```

Comment:
Do we want to return even before conditionally setting `Content-Length`? (I didn't mention `Date` as it's going  away in #7392).